Custom-managed identity (#3875)
astronomer/docs: astro/authorize-deployments-to-your-cloud.md

   ```json
   {
     "Version": "2012-10-17",
     "Statement": [
       {
         "Effect": "Allow",
         "Principal": {
           "AWS": ["<workload-identity-role>"]
         },
         "Action": "sts:AssumeRole"
       }
     ]
   }
   ```

7. Click **Update policy**.

Repeat these steps for each Astro Deployment that needs to access your AWS resources.

### Step 2: Create an Airflow connection

Now that your Deployment is authorized, you can connect it to your cloud using an Airflow connection. Either create an **Amazon Web Services** connection in the [Astro UI](create-and-link-connections.md) or the Airflow UI for your Deployment and specify the following fields:

- **Connection Id**: Enter a name for the connection.
- **Extra**:

  ```json
  {
    "role_arn": "<your-role-arn>",
    "region_name": "<your-region>"
  }
  ```

If you don't see **Amazon Web Services** as a connection type in the Airflow UI, ensure you have installed its provider package in your Astro project's `requirements.txt` file. See **Use Provider** in the [Astronomer Registry](https://registry.astronomer.io/providers/Amazon/versions/latest) for the latest package.

> ### Custom IAM identity

Author: ```suggestion
### Attach an IAM role to your Deployment
```

:::publicpreview
:::

> #### Prerequisites

Reviewer: Add "new or existing IAM role with required permissions to your data sources"

> #### Prerequisites

Reviewer: May want to also add:
If using a local terminal the **aws**, **jq**, and **openssl** CLIs are required. If using AWS Cloud Shell,
the required CLIs are enabled by default.

- Minimum Astro Runtime version:
    - 9.15.0
    - 10.9.0
    - 11.5.0

> #### Step 1: Authorize the Deployment to your cloud

Reviewer: ```suggestion
#### Step 1: Authorize the Deployment to your IAM role
```

> You can now assign a workload identity, specifically an AWS IAM role, to your Airflow Deployments on Astro. The Deployment and configured Airflow connections authorize the Deployment's access to customer data services using the customer managed identity. This adds the following benefits:

Reviewer: We should keep the messaging similar for AWS and GCP. Suggest adding this same blurb to GCP doc.

Author: Even though GCP uses service account and not IAM role?

Reviewer: That terminology would differ but otherwise the steps are the same.

Author: Since I've included basically this positioning in the release notes, going to mirror the GCP intro here for consistency and remove redundancy between release notes and docs

    - Leverage existing identities when migrating from MWAA or OSS environments
    - Reduced migration friction and change risk

To authorize your Deployment, create an IAM role that is assumed by the Deployment's workload identity:

1. In the Astro UI, select your Deployment and then click **Details**. Copy the Deployment's **Workload Identity**.
2. In the Airflow UI, click the **Admin** tab and go to you **Configurations** page. Copy the **Deployment Namespace**.
3. In the AWS account that contains your AWS service, create an IAM role. See [Creating a role to delegate permissions to an AWS service](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html).
4. In the AWS Management Console, go to the Identity and Access Management (IAM) dashboard.
5. Click **Roles** and in the **Role name** column, select the role you created in Step 2.
6. Click **Trust relationships**.
7. Click **Edit trust policy** and paste the workload identity you copied from Step 1 and the Deployment namespace from Step 2 in the trust policy. Your policy should look like the following:

> 

Reviewer: dont need to show this block, users can get the entire script from the UI

```json
    {
    "Effect": "Allow",
    "Principal": {
        "Federated": "<your-workload-identity-role"
    },
    "Action": "sts:AssumeRoleWithWebIdentity",
    "Condition": {
        "StringLike": {
            "<clusterOIDCIssuerUrl>:aud": "sts.amazonaws.com",
            "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:<DeploymentNamespace>:*"
        }
    }
    }
```

:::tip

> If you want to enable custom managed identities on ephemeral Deployments, change the `<DeploymentNamespace>` value in `Condition` to include a wildcard. The following shows an example:

Reviewer: ```suggestion
If you want to share or re-use the same customer managed identity on static or ephemeral Deployments, without having to update your Trust Policy for every net new Deployment, change the `<DeploymentNamespace>` value in `Condition` to include a wildcard. The following shows an example:
```

> 

Reviewer: I would also suggest we put a red Warning here stating something like:
"Only perform this action if creating Deployments on a Dedicated Cluster belonging to your Astro Organization"

Author: I can't put a warning note inside of a tip, but I added a bunch of DEDICATED CLUSTER ONLY to the note

    ```json
    {
        "Condition": {
            "StringLike": {
                "<clusterOIDCIssuerUrl>:aud": "sts.amazonaws.com",
                "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:*:*"
            }
        }
    }
    ```
:::

> 7. Click **Update policy**.

Reviewer: remove step 7

Repeat these steps for each Astro Deployment that needs to access your AWS resources.

#### Step 2: Create an Airflow connection

Now that your Deployment is authorized, you can connect it to your cloud using an Airflow connection. Either create an **Amazon Web Services** connection in the [Astro UI](create-and-link-connections.md) or the Airflow UI for your Deployment and specify the following fields:

- **Connection Id**: Enter a name for the connection.
- **Extra**:

  ```json
  {

>     "role_arn": "<your-role-arn>",

Reviewer: can be removed... no longer required with customer managed identity

- **Connection Id**: Enter a name for the connection.
- **Extra**:

  ```json
  {
    "role_arn": "<your-role-arn>",
    "region_name": "<your-region>"
  }
  ```

If you don't see **Amazon Web Services** as a connection type in the Airflow UI, ensure you have installed its provider package in your Astro project's `requirements.txt` file. See **Use Provider** in the [Astronomer Registry](https://registry.astronomer.io/providers/Amazon/versions/latest) for the latest package.

### Attach an IAM role to your Deployment

<HostedBadge/>

:::publicpreview
:::

You can attach an AWS IAM role to your Deployment to grant the Deployment all of the role's permissions.

> Using IAM roles provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing IAM roles on new Deployments, or your can attach a single IAM role to multiple Deployments that all have the same level of access to your cloud.

Reviewer 2: maybe say "...that all require the same level of access to your cloud"

> Using IAM roles provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing IAM roles on new Deployments, or your can attach a single IAM role to multiple Deployments hat all require the same level of access to your cloud.

Reviewer: ```suggestion
Using IAM roles provides the greatest amount of flexibility for authorizing Deployments to your cloud. For example, you can use existing IAM roles on new Deployments, or your can attach a single IAM role to multiple Deployments that all require the same level of access to your cloud.
```

#### Prerequisites

- Minimum Astro Runtime version:
    - 9.15.0
    - 10.9.0
    - 11.5.0
- A new or existing IAM role in your data sources with the required permissions you want your Deployment to have.
- If using [AWS CloudShell](https://aws.amazon.com/cloudshell/),
the required CLIs are enabled by default.
- If you use a local terminal, the following CLIs are required:
  - [AWS CLI](https://aws.amazon.com/cli/)
  - [jq](https://jqlang.github.io/jq/)
  - [openSSL](https://www.openssl.org/source/)

#### Step 1: Authorize the Deployment to your IAM role

To authorize your Deployment, create an IAM role to assign as your Deployment's workload identity:

> 1. [Create an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html) to delegate permissions to in an AWS service. Grant the role any permission that the Deployment will need in your AWS account. Copy the IAM role ID to use later in this setup.

Reviewer: ```suggestion
1. [Create an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html) to delegate permissions to in an AWS service. Grant the role any permission that the Deployment will need in your AWS account. Copy the IAM role ARN to use later in this setup.
```

2. In the Astro UI, select your Deployment and then click **Details**. In the **Advanced** section, click **Edit**.
3. In the **Workload Identity** menu, select **Customer Managed Identity**.

> 4. Enter your IAM role ID when prompted, then copy and run the provided CLI command.

Reviewer: ```suggestion
4. Enter your IAM role ARN when prompted, then copy and run the provided CLI command.
```

> 5. Click **Update Deployment**. The service account is now selectable as a workload identity for the Deployment.

Reviewer: ```suggestion
5. Click **Update Deployment**. The IAM role is now selectable as a workload identity for the Deployment.
```

Repeat these steps for each Astro Deployment that needs to access your AWS resources.

:::tip Dedicated Cluster only

If you want to share or re-use the same customer managed identity on static or ephemeral Deployments for dedicated clusters, without having to update your Trust Policy in your AWS account for every net new Deployment, change the `<DeploymentNamespace>` value in `Condition` to include a wildcard. The following shows an example:

    ```json
    {
        "Condition": {
            "StringLike": {
                "<clusterOIDCIssuerUrl>:aud": "sts.amazonaws.com",
                "<clusterOIDCIssuerUrl>:sub": "system:serviceaccount:*:*"
            }
        }
    }
    ```
:::

#### Step 2: Create an Airflow connection

> Now that your Deployment is authorized, you can connect it to your cloud using an Airflow connection. Either create an **Amazon Web Services** connection in the [Astro UI](create-and-link-connections.md) or the Airflow UI for your Deployment and specify the following fields:

Reviewer 2: Maybe it should be: "Create an ... connection either in the ..."
